app/models/shipping_co.rb

```diff
@@ -1,10 +1,16 @@
 class ShippingCo < ApplicationRecord
   has_many :price_settings, dependent: :destroy
   has_many :employees, dependent: :restrict_with_error
+  has_many :reach_areas, dependent: :destroy
 
   validates :name, :corp_name, :cnpj, :address, :status, presence: true
   validates :name, :corp_name, :cnpj, uniqueness: true
   validates :cnpj, length: { is: 14 }
 
   enum status: { inactive: 0, active: 1 }
+
+  def self.available_to_customer(region:)
+    covered_region = ReachArea.where(region: region)
+    covered_region.map { |state| state.shipping_co if state.shipping_co.active? }
```

Comment:
```suggestion
  ShippingCo.active.joins(:reach_areas).where(reach_areas: { region: region })
```

Essa talvez seja uma das poucas vezes que a gente falou disso então vou detalhar, mas quando vamos fazer esse tipo de verificação se der pra pedir pro banco de dados fazer sempre é melhor. Porque é uma aplicação feita pra ler uma massa de dados e calcular pra gente, isso é bem mais performático que o Rails fazer (normalmente e dependendo da massa de dados) e evita queries de N+1.

Nesse caso como o resultado esperado é ShippingCompany podemos começar com elas, verificar as ativas e fazer um join na tabela `reach_areas` para poder perguntar pra essa segunda tabera e trazer somente os dados que batem nessas condições.

Reply:
Nossa, estavamos quebrando a cabeça tentando chegar nessa solução. Com certeza fica muito melhor assim. :D